moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1170,8 +1170,13 @@ void TrajectoryExecutionManager::stopExecution(bool auto_clear)
       ROS_INFO_NAMED(name_, "Stopped trajectory execution.");
 
       // wait for the execution thread to finish
-      execution_thread_->join();
-      execution_thread_.reset();
+      execution_thread_mutex_.lock();
```

Comment:
Please use a `scoped_lock` to ensure automatic unlocking of the mutex when leaving the context instead of manual locking and unlocking.